crates/ide/src/join_lines.rs

```diff
@@ -990,6 +1034,55 @@ fn main() {
 
     }
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn join_assignments() {
+        check_join_lines(
+            r#"
+fn foo() {
+    $0let foo;
+    foo = "bar";
+}
+"#,
+            r#"
+fn foo() {
+    $0let foo = "bar";
+}
+"#,
+        );
+
+        cov_mark::check!(join_assignments_mismatch);
+        check_join_lines(
+            r#"
+fn foo() {
+    let foo;
+    let qux;$0
+    foo = "bar";
+}
+"#,
+            r#"
+fn foo() {
+    let foo;
+    let qux;$0 foo = "bar";
+}
+"#,
+        );
+
+        cov_mark::check!(join_assignments_already_initialized);
```

Comment:
:heart: 

I didn't realise that you can put several marks in a single block and have them do what you want them to!